client/main.lua

```diff
@@ -125,13 +123,15 @@ function loadAnimDict(dict)
 end
 
 function IsAuthorized(doorID)
-	for _,job in pairs(doorID.authorizedJobs) do
-		if job == PlayerJob.name or job == PlayerGang.name then
-			return true
-		end
-	end
-	
-	return false
+    for _, job in pairs(doorID.authorizedJobs.jobs) do
+        if PlayerJob.name == job.name and PlayerJob.grade.level >= job.grade then
+            return true
+        elseif PlayerGang.name == job.name and Playergang.grade.level >= job.grade then
```

Comment:
Playergang should be PlayerGang.

Reply:
We can probably reduce this to one if statement, instead of two, don't you agree?